yoyodyne/models/hard_attention.py

```diff
@@ -503,3 +426,189 @@ def add_argparse_args(parser: argparse.ArgumentParser) -> None:
             help="Width of attention context "
             "(hard attention architectures only). Default: %(default)s.",
         )
+
+
+class HardAttentionGRUModel(HardAttentionRNNModel):
```

Comment:
Hmm, this redundancy of boilertplate strikes me as unnecessary. Is there any way we can just keep the underlying decoder structure as abstracted and just pass a flag that allows the attention decoder to specify architecture in module construction?

Reply:
I tried and it’s nearly impossible because GRUs don’t return cell state and LSTMs do, and you can’t just ignore it.

Reply:
Hmm, aight. I think this is a case for a future PR where we just use GRU/LSTM cells classes then.